services/director/docker/boot.sh

```diff
@@ -27,14 +29,18 @@ then
   which python | sed 's/^/    /'
   echo $INFO "PIP :"
   $SC_PIP list | sed 's/^/    /'
+
+  #------------
+  echo $INFO "setting entrypoint to use watchmedo autorestart..."
+  entrypoint='watchmedo auto-restart --recursive --pattern="*.py" --'
 fi
 
 # RUNNING application ----------------------------------------
 if [[ ${SC_BOOT_MODE} == "debug-ptvsd" ]]
 then
   echo
   echo $INFO "PTVSD Debugger initializing in port 3004"
-  python3 -m ptvsd --host 0.0.0.0 --port 3000 -m \
+  eval "$entrypoint" python3 -m ptvsd --host 0.0.0.0 --port 3000 -m \
```

Comment:
What ``eval`` does actually do? We need to be sure you are NOT creating a subprocess. Otherwise, kill functions will not be transmitted to your program and therefore the director has no chance to exit gracefully. Have a look at this article [this article](https://kkc.github.io/2018/06/06/gracefully-shutdown-docker-container/#Process-behavior-in-Docker). 

The test you disabled is suppose to check this ... so hopefully if something went wrong, we should be able to detect it

IN any case, the way I check is 
a) ``docker exec `` your container with a shell and ``ps aux`` to see how many processes are actually running in the docker

Reply:
to be honest I don't know, but this is for development purpose only, not in production mode isn't it?